src/Traits/AllowedLinkClassesTrait.php

```
@@ -0,0 +1,117 @@
+                $validClasses[] = $type;
+            }
+        }
+
+        return count($validClasses) > 0;
+    }
+
+    /**
+     * Set disabled types for LinkField
+     */
+    public function setDisabledTypes(array $types): static
+    {
+        if (is_array($types) || !empty($types)) {
+            $this->allowed_types = array_diff($this->genarateAllowedTypes(), $types);
+        }
+
+        return $this;
+    }
+
+    public function getAllowedTypes(): array
+    {
+        return $this->allowed_types;
+    }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function getTypesProps(): string
+    {
+        return json_encode($this->list());
+    }
+
+    /**
+     * @return Link[]
+     * @throws InvalidArgumentException
+     */
+    public function list(): array
+    {
+        $types = [];
+        $typeDefinitions = $this->genarateAllowedTypes();
+
+        foreach ($typeDefinitions as $key => $class) {
+            $types[$key] = LinkTypeService::definitionToType($class);
+        }
+
+        $newArray = [];
+
+        array_map(function (Link $type, string $key) use (&$newArray) {
+            $newArray[$key] = [
+                'key' => $key,
+                'title' => $type->LinkTypeTile(),
+                'handlerName' => $type->LinkTypeHandlerName(),
+            ];
+        }, $types, array_keys($types));
+
+        return $newArray;
+    }
+
+    /**
+     * Generate allowed types with key => value pair
+     * Example: ['cms' => SiteTreeLink::class]
+     */
+    private function genarateAllowedTypes(array $types = []): array
+    {
+        $typeDefinitions = !empty($types) ? $types : $this->getAllowedTypes();
+
+        if (empty($typeDefinitions)) {
+            $typeDefinitions = ClassInfo::subclassesFor(Link::class);
+        }
+
+        $result = array();
+        foreach ($typeDefinitions as $class) {
+            if (is_subclass_of($class, Link::class)) {
```

Seems to me that this should throw an exception if you give it something that doesn't exists.

---

I decided that it would be better if a complete list of all types of links was returned if the user passed something wrong or not a subclass of the Link class as a parameter.

---

If you're passing something that doesn't map to an actual Link DataObject, there's a high likelihood that there's a mistake in your code.

Logically, we should throw an exception in that scenario.